### ISSUE TITLE
captureHTTPsGlobal fails when bundled with esbuild

### ISSUE THREAD

**Issue by christophgysin** [aws/aws-xray-sdk-node#433]
When using AWS CDK to deploy Lambda functions with [aws-lambda-nodejs](https://docs.aws.amazon.com/cdk/api/latest/docs/aws-lambda-nodejs-readme.html), trying to call:
```
xray.captureHTTPsGlobal(http, false)
```
from aws-xray-sdk-node>=3.3.2 causes the function to fail with:
```
Cannot set property request of #<Object> which has only a getter
```
This is caused by [esbuild](https://github.com/evanw/esbuild) which aws-lambda-nodejs uses to bundle the code. For details, see: https://github.com/evanw/esbuild/issues/587

Now while this is really a bug in the bundler, it would be easy to fix this in aws-xray-sdk-node, and I'd be happy to make a PR.

**Comment by srprash** [comment]
Hi @christophgysin 
Thanks for submitting the fix for this issue. I'm not familiar with esbuild, so is there something that aws-lambda-nodejs needs to change as well?

**Comment by christophgysin** [comment]
@srprash No changes are needed in aws-lambda-nodejs.

**Comment by teemuniiranen** [comment]
I encountered this with aws-sdk 2.0.0-rc.11 and the latest aws-xray-sdk 3.3.3. In what release this is included? Many thanks!

**Comment by cward-extron** [comment]
@willarmiros  In which release is this fix included? It looks like the latest (3.3.3) was released on May 13th but this fix was added on June 1st..
